### DISCUSSION TITLE
Is there an appetite for cmake based builds?

### DISCUSSION THREAD

**Discussion by nolan-veed** [roc-streaming/roc-toolkit#586] [Discuss]
I've embarked on a journey to create a cmake based build, at least for development purposes, so that I can use my IDE CLion, which doesn't support SCons.

So far, I'm successful with at leat building the roc_core module and it's tests allowing me to start writing code.

Just wondering, whether it'll be is useful moving forward, at least to partially support this build mechanism.

Nolan

**Comment by gavv** [comment]
Hi,

What kind of support are you missing from IDE?

I think it should be enough to configure clion to use custom build command and to use compilation database generated by scons (compile_commands.json). I think it should work with any modern ide or editor.

Supporting two build systems is something that I'd like to avoid, at least without strong reasons. It's a big burden.

Also, while scons requires more work, it gives a much cleaner and nicer result. And since that additional work is already done in roc's scons plugin, so far I see no big reasons to switch.

**Reply by nolan-veed** [reply]
Ah! I see - the build/run configs didn't get discovered automatically with the compilation database. Let me try again, and also see if I can easily add custom commands, and whether I can debug individual tests, for example. Al of which, should be doable... I think it's just the convenience. 

May be there's a happy middle ground, where SCons acts like a wrapper around cmake stuff.
